k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -105,270 +103,277 @@ class SingleNsNamer(
   }
 }
 
+object SingleNsNamer {
+  protected val PrefixLen = 2
+}
+
 abstract class EndpointsNamer(
   idPrefix: Path,
-  mkApi: String => NsApi,
-  backoff: Stream[Duration] = Backoff.exponentialJittered(10.milliseconds, 10.seconds)
-)(implicit timer: Timer = DefaultTimer) extends EnumeratingNamer {
+  mkApi: String => v1.NsApi,
+  labelName: Option[String] = None,
+  backoff: Stream[Duration] = EndpointsNamer.DefaultBackoff
+)(implicit timer: Timer = DefaultTimer)
+  extends Namer with Stabilize {
 
   import EndpointsNamer._
+  protected[this] val variablePrefixLength: Int
```

Comment:
the concrete namer implementations [use it](https://github.com/linkerd/linkerd/pull/1587/files/6b648dbf421303b3f483e0f843aaf4d50c77116e#diff-6e1db8d78e58df54b2256d8d193c5000R37), I moved it to the `EndpointsNamer` class because I was planning on trying to factor out more of `lookup` from the concrete classes.